Bluejay.asm

```diff
@@ -341,14 +342,14 @@ Temp_Storage:				DS	48	; Temporary storage (internal memory)
 ;**** **** **** **** ****
 ; EEPROM code segments
 ; A segment of the flash is used as "EEPROM", which is not available in SiLabs MCUs
-IF MCU_48MHZ == 2
+IF MCU_TYPE == 2
 	CSEG AT 3000h
 ELSE
 	CSEG AT 1A00h
 ENDIF
 EEPROM_FW_MAIN_REVISION		EQU	0	; Main revision of the firmware
 EEPROM_FW_SUB_REVISION		EQU	18	; Sub revision of the firmware
-EEPROM_LAYOUT_REVISION		EQU	204	; Revision of the EEPROM layout
+EEPROM_LAYOUT_REVISION		EQU	205	; Revision of the EEPROM layout
```

Comment:
For BB51 ref voltage is always the same 1.65V, but for BB21 it depends on whether we are using 1S (1.65V) or 2S+ (vdd). For that reason we would need the new Pgm_Power_Rating parameter and the new layout revision.

Reply:
Sorry, I got confused there - already had the other file collapsed and thought you just renamed a variable.

What happened to `_Pgm_Throttle_Rate`?

With additional variables it is important that we add them at the end of the EEPROM section (or at least a previously unused address), otherwise we might run into issues when migrating settings from one version to the next.

In this case you are replacing one setting with another. I am not 100% sure what the implications are here, so I would like to continue expanding on the current "EEPROM" layout.

What is now a bit shitty is, that 205 has already been used inbetween for some other settings that only made it in pre-releases, PWM, beacon melody and such.

This means we would need to go with 206 here, which inherits properties from 204. As reference, have a look here how the "EEPROM" is mapped:
- [offsets](https://github.com/stylesuxx/esc-configurator/blob/master/src/sources/Bluejay/eeprom.js)
- [setting versions](https://github.com/stylesuxx/esc-configurator/blob/master/src/sources/Bluejay/settings.js)

In the configurator I would then implement it as an ENUM instead of a checkbox, then we can keep the numerical values

Reply:
Ok, I will implement it in a new parameter after the existing ones

Reply:
Umm it seems that Pgm_Pwm_Freq already exists, so I will use it with its possible values 24 (or any other value), 48, 96, in a new pw when to implement PWM setting.

For the moment I won't change the current pr

Reply:
Updated the pr:
- Moved Eep_Pgm_Power_Rating to be the last parameter.
- Fixed erase_and_store_all_in_eeprom to also rewrite Eep_Pgm_Power_Rating parameter.
- Added beeps at the end of erase_and_store_all_in_eeprom to signal action done and giving time to flash controller to settle data down in flash

Reply:
We will need to change the layout revision in any case. There are people out there that are using revision 205 right now, so we can not add to this.

The question is just if 206 will inherit settings from 204 or 205. If it inherits from 205 we need to implement all the other features (variable PWM, melody as beacon,...) that are present in those settings, otherwise we could easily roll out a new version with only this feature and then merge them back together in 207.

I have no preference here, just mentioning it.

Reply:
If we know 205 layout we could use 206 for this new one, simply adding the new setting to 205 layout and creating 206 layout. Do you have any info about layout 205?

Reply:
Yes, but we would then also need to implement the features of 205 layout before we push this one out, that's why I think that it would be best to got from 204->206 and then when we merge the 205 features we can merge 206 and 205 into 207.

[205 contains](https://github.com/stylesuxx/esc-configurator/blob/e23a3e0cd89d5b1af251579d47e1bae151e3a393/src/sources/Bluejay/settings.js#L448) settings for "variable" PWM, and beacon melody, basically the features of #18 and  #19

Reply:
I understand. Then I will bump version to 206

Reply:
EEPROM layout bumped to 206

@stylesuxx When you do the integration, please, could you integrate first edt_events? edt_events scheduler is newer and publishes more signals. For me will be easier to fix this branch's conflicts after integrating edt_events.